src/WebJobs.Script.WebHost/Diagnostics/NonDisposableMetricsLogger.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    /**
+     * This decorator is required to enforce a single instance of WebHostMetricsLogger across webhost and jobhost and also not letting it get disposed during jobhost restarts. The default DryIoc implementation disposes instances being reused.
```

Comment:
This comment may lead someone without context, who knows the DI framework supports that option, to try to "clean this up" by enabling disposal suppression and removing this decorator. 
I would change this:
> The default DryIoc implementation disposes instances being reused

to something like:
`The host DI implementation disposes of registered instances, which is behavior that cannot be changed`